discovery_engine_core/core/src/engine.rs

```diff
@@ -347,29 +347,17 @@ where
         let markets = self.config.markets.read().await;
         let mut stacks = self.stacks.write().await;
 
-        let mut errors = vec![];
-        for market in markets.iter() {
-            if let Err(error) = update_stacks(
-                &mut stacks,
-                &mut self.exploration_stack,
-                &mut self.ranker,
-                history,
-                self.core_config.take_top,
-                self.core_config.keep_top,
-                request_new,
-                market,
-            )
-            .await
-            {
-                errors.push(error);
-            }
-        }
-
-        if errors.is_empty() {
-            Ok(())
-        } else {
-            Err(Error::Errors(errors))
-        }
+        update_stacks(
+            &mut stacks,
+            &mut self.exploration_stack,
+            &mut self.ranker,
+            history,
+            self.core_config.take_top,
+            self.core_config.keep_top,
+            request_new,
+            &*markets,
```

Comment:
:art: the additional deref is not needed for inference to work
```suggestion
            &markets,
```